sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs

```diff
@@ -27,318 +25,295 @@
+        void AddKey(object hashKey);
 
         /// <summary>
-        /// Executes a server call to batch-get the items requested.
-        /// Populates Results with the retrieved items.
+        /// Add a single item to get, identified by its hash-and-range primary key.
         /// </summary>
-        internal protected abstract void ExecuteHelper();
+        /// <param name="hashKey">Hash key of the item to get</param>
+        /// <param name="rangeKey">Range key of the item to get</param>
+        void AddKey(object hashKey, object rangeKey);
 
-#if AWS_ASYNC_API
         /// <summary>
-        /// Executes an asynchronous server call to batch-get the items requested.
-        /// Populates Results with the retrieved items.
+        /// Add a single item to get.
         /// </summary>
-        internal protected abstract Task ExecuteHelperAsync(CancellationToken cancellationToken);
-#endif
-        #endregion
+        /// <param name="keyObject">Object key of the item to get</param>
+        void AddKey(T keyObject);
 
+        /// <summary>
+        /// Creates a MultiTableBatchGet object that is a combination
+        /// of the current BatchGet and the specified BatchGets
+        /// </summary>
+        /// <param name="otherBatches">Other BatchGet objects</param>
+        /// <returns>
+        /// An interface with the ability to perform MultiTableBatchGet operations.
+        /// </returns>
+        IMultiTableBatchGet Combine(params IBatchGet[] otherBatches);
+    }
 
-        #region Internal methods
+    /// <summary>
+    /// Represents a non-generic object for retrieving a batch of items
+    /// from a single DynamoDB table
+    /// </summary>
+    public abstract class BatchGet
+    {
+        internal DocumentBatchGet DocumentBatch { get; set; }
 
         internal abstract void CreateDocumentBatch();
-        internal abstract void PopulateResults(List<Document> items);
 
-        #endregion
+        internal abstract void PopulateResults(List<Document> items);
     }
 
     /// <summary>
     /// Represents a strongly-typed object for retrieving a batch of items
     /// from a single DynamoDB table
     /// </summary>
-    public class BatchGet<T> : BatchGet
+    public partial class BatchGet<T> : BatchGet, IBatchGet<T>
     {
-        #region Public properties
+        private readonly DynamoDBContext _context;
+        private readonly DynamoDBFlatConfig _config;
+        private readonly ItemStorageConfig _itemStorageConfig;
+        private readonly List<Key> _keys;
 
-        /// <summary>
-        /// List of results retrieved from DynamoDB.
-        /// Populated after Execute is called.
-        /// </summary>
-        new public List<T> Results { get { return TypedResults; } }
+        /// <inheritdoc/>
+        public int TotalKeys => _keys.Count;
 
-        #endregion
+        /// <inheritdoc/>
+        public List<object> UntypedResults { get; private set; } = new List<object>();
```

Comment:
This property should be on the base class `BatchGet` that implements `IBatchGet` the non-generic versions.

Reply:
Done